src/components/inputs/TokenInput/TokenInput.vue

```diff
@@ -203,8 +205,12 @@ const decimalLimit = computed<number>(() => token.value?.decimals || 18);
 /**
  * METHODS
  */
-function handleAmountChange(amount: InputValue) {
+function handleAmountChange(amount: InputValue, isSetMax = false) {
   const safeAmount = overflowProtected(amount, decimalLimit.value);
+  if (isSetMax) {
+    emit('setMax', safeAmount);
+    return;
+  }
```

Comment:
I implemented this because of `handleTokenSwitch` function. now when user switches tokens, `exactIn` value is set improperly due to the sideEffects of token switching complicated logic (it changes twice)

Reply:
I still don't understand why we would need to add a setMax event because of a token switch event. Could we not fix the underlying issue? Why are things changing twice? If you think it might help, you could record a loom going through the interaction and the code to show what the actual issue is?